homeassistant/components/upc_connect/device_tracker.py

```diff
@@ -95,12 +100,38 @@ def __init__(self, hass, config):
 
             self.token = response.cookies["sessionToken"].value
 
-            return True
+            return await self._async_initialize_token_with_password(CMD_LOGIN)
 
         except (asyncio.TimeoutError, aiohttp.ClientError):
             _LOGGER.error("Can not load login page from %s", self.host)
             return False
 
+    async def _async_initialize_token_with_password(self, function):
+        """Get token with password."""
+        try:
+            with async_timeout.timeout(10):
+                response = await self.websession.post(
```

Comment:
Please extract the device specific interface code, I/O and data parsing, to a standalone library published on pypi.

https://developers.home-assistant.io/docs/en/creating_platform_code_review.html#4-communication-with-devices-services